doc/design/context.md

```diff
@@ -0,0 +1,171 @@
+## Context Design
+
+`Net` is the container and controller of a set of `Operator`. Each `Operator` in `Net` has a method called `Run` to make computation. The `Run` method of `Operator` is defined as follows:
+
+```c++
+Error Operator::Run(OpContext* context);
```

Comment:
What kind of errors could `Operator::Run` return? What actions could the caller take in response to these errors?

I ask because I always tend to handle errors inside the function instead of leaving errors to the caller, who has less information as the callee does and usually cannot do much with the returned errors.